addons/mass_mailing_crm/models/mass_mailing.py

```diff
@@ -48,13 +49,17 @@ def action_redirect_to_opportunities(self):
         return action
 
     def _get_crm_utm_domain(self):
-        res = []
+        """ We want both archived and active records that match the UTMs """
+        domain = []
         if self.campaign_id:
-            res.append(('campaign_id', '=', self.campaign_id.id))
+            domain = expression.AND([domain, [('campaign_id', '=', self.campaign_id.id)]])
         if self.source_id:
-            res.append(('source_id', '=', self.source_id.id))
+            domain = expression.AND([domain, [('source_id', '=', self.source_id.id)]])
         if self.medium_id:
-            res.append(('medium_id', '=', self.medium_id.id))
-        if not res:
-            res.append((0, '=', 1))
-        return res
+            domain = expression.AND([domain, [('medium_id', '=', self.medium_id.id)]])
+        if not domain:
+            domain = expression.AND([domain, [(0, '=', 1)]])
+        else:
+            domain = expression.AND([domain, ['|', ('active', '=', True), ('active', '=', False)]])
```

Comment:
Shouldn't we use the active_test context key when performing the search instead ? That way that's the caller who decide whether to include active records or not, not the domain itself that should only be based on UTM data. What do you think about that ?

Reply:
@tde-banana-odoo Good point, I've updated the code accordingly!